File: src/main/java/ru/ezhov/rocket/action/api/RocketActionConfiguration.java
package ru.ezhov.rocket.action.api;

import java.util.List;

/**
 * Configuring actions
 */
public interface RocketActionConfiguration {
    /**
     * @return configurable action type
     */
    String type();

    /**
     * @return configurable action description
     */
    String description();

    /**
     * @return list of action properties to configure
     */
    List<RocketActionConfigurationProperty> properties();
}

File: src/main/kotlin/ru/ezhov/rocket/action/api/RocketActionConfiguration.kt
package ru.ezhov.rocket.action.api

/**
 * Конфигурация действия
 */
interface RocketActionConfiguration {
    /**
     * Уникальное значение в рамках всех загруженных действия
     *
     * @return тип действия
     */
    fun type(): RocketActionType

    /**
     * Название действия для отображения
     *
     * @return название
     */
    fun name(): String

    /**
     * Описание действия для отображения
     *
     * @return описание
     */
    fun description(): String

    /**
     * Свойства действия для конфигурирования
     *
     * @return свойства действия
     */
    fun properties(): List<RocketActionConfigurationProperty>
}